how-to/web-interop-support-context-and-intents/client/src/platform/layout/layout-override.ts

```diff
@@ -0,0 +1,258 @@
+ * @returns a function call.
+ */
+export function makeOverride(
+	fin: OpenFin.Fin<OpenFin.EntityType>,
+	layoutContainerId: string,
+	layoutSelectorId: string
+) {
+	return function layoutManagerOverride(Base: LayoutManagerConstructor): LayoutManagerConstructor {
+		/**
+		 * @class LayoutManagerBasic
+		 * This implementation is the fundamental override for Multiple Layouts in Web.
+		 */
+		return class LayoutManagerBasic extends Base implements LayoutManager {
+			private _layoutMapArray: LayoutManagerItem[];
+
+			private readonly _layoutContainer: HTMLElement | null;
+
+			private readonly _layoutSelector: HTMLSelectElement | null;
+
+			private _selectedLayout: string | undefined;
+
+			/**
+			 * Sets up the required variables for the layout manager.
+			 */
+			constructor() {
+				super();
+				this._layoutMapArray = [];
+				this._layoutContainer = document.querySelector<HTMLElement>(`#${layoutContainerId}`);
+				this._layoutSelector = document.querySelector<HTMLSelectElement>(`#${layoutSelectorId}`);
+				if (this._layoutSelector !== null) {
+					this._layoutSelector.addEventListener("change", async (event) => {
+						const selectElement = event.target as HTMLSelectElement;
+						const selectedValue = selectElement.value;
+						await this.showLayout({ layoutName: selectedValue, uuid: fin.me.uuid, name: fin.me.name });
+					});
+				}
+			}
+
+			/**
+			 * Our implementation of getLayoutSnapshot that contains titles.
+			 * @returns Promise<PlatformLayoutSnapshot>
+			 */
+			public async getLayoutSnapshot(): Promise<PlatformLayoutSnapshot> {
+				const layoutSnapshot = await super.getLayoutSnapshot();
+				const platformLayoutSnapshot: PlatformLayoutSnapshot = {
+					layouts: layoutSnapshot.layouts,
+					layoutTitles: {},
+					layoutSelected: this._selectedLayout
+				};
+				for (const layout of this._layoutMapArray) {
+					if (layout.layoutTitle !== undefined) {
+						platformLayoutSnapshot.layoutTitles[layout.layoutName] = layout.layoutTitle;
+					}
+				}
+				return platformLayoutSnapshot;
+			}
+
+			/**
+			 * Override for applying multiple snapshots.
+			 * @param snapshot The layouts object containing the fixed set of available layouts.
+			 */
+			public async applyLayoutSnapshot(snapshot: WebLayoutSnapshot): Promise<void> {
+				console.log(`[Apply Layout] Does this exist? ${Boolean(this._layoutContainer)}`);
+				if (this._layoutContainer !== null && this._layoutContainer !== undefined) {
+					const platformLayoutSnapshot = snapshot as PlatformLayoutSnapshot;
+					for (const [key, value] of Object.entries(snapshot.layouts)) {
+						const layoutTitle =
+							platformLayoutSnapshot?.layoutTitles === undefined
+								? undefined
+								: platformLayoutSnapshot.layoutTitles[key];
+						this._layoutMapArray.push({
+							layoutName: key,
+							layoutTitle,
+							layout: value,
+							container: this._layoutContainer
+						});
+					}
+					setTimeout(async () => {
```

Comment:
why the set timeout?

Reply:
This is taken from the layout sample where Joe had conversations with Pierre and there was a reason why it was needed. When Joe is back from holiday we can touch upon it with you and Pierre to see if we can strip it from the samples that have that.